Makefile

```diff
@@ -1,25 +1,80 @@
-setup:
-	curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-	pyenv install '2.7.14'
-	pyenv local 2.7.14
-	pyenv virtualenv 2.7.14 samtranslator27
-	pyenv activate samtranslator27
-	pyenv local samtranslator27
+NAME=samtranslator27
+PYTHON_VERSION=2.7.14
+CODE_COVERAGE=95
+PYENV := $(shell command -v pyenv 2> /dev/null)
+
+target:
+	$(info ${HELP_MESSAGE})
+	@exit 0
+
+# Make sure that pyenv is configured properly and that we can use it in our setup target.
+validation:
+ifndef PYENV
+    $(error "make sure pyenv is accessible in your path, (usually by adding to PATH variable in bash_profile, zshrc, or other locations based on your platform) See: https://github.com/pyenv/pyenv#installation for the installation insructions.")
+endif
+ifndef PYENV_SHELL
+	$(error "Add 'pyenv init' to your shell to enable shims and autocompletion, (usually by adding to your bash_profile, zshrc, or other locations based on your platform)")
+endif
+ifndef PYENV_VIRTUALENV_INIT
+	$(error "Add 'pyenv virtualenv-init' to your shell to enable shims and autocompletion, (usually by adding to your bash_profile, zshrc, or other locations based on your platform)")
+endif
+
+install:
+	$(info [*] Install pyenv using https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer...)
+	@curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer 2> /dev/null | bash
+
+setup: validation
+	$(info [*] Download and install python $(PYTHON_VERSION)...)
+	@pyenv install $(PYTHON_VERSION)
+	@pyenv local $(PYTHON_VERSION) 
+	$(info [*] Create virtualenv $(NAME) using python $(PYTHON_VERSION)...)
+	@pyenv virtualenv $(PYTHON_VERSION) $(NAME)
+	@$(MAKE) activate
+	
+activate:
+	$(info [*] Activate virtualenv $(NAME)...)
+	$(shell eval "$$(pyenv init -)" && eval "$$(pyenv virtualenv-init -)" && pyenv activate $(NAME) && pyenv local $(NAME))
+	
+clean:
```

Comment:
I know this is convenient to clean things, but can we actually remove it from here? This might work for someone setting up SAM Translator with pyenv and other things, but they are by no means required to. I am concerned that we are going beyond the scope.

Thoughts?